kmm/kmm-feed/src/commonMain/kotlin/com/gchristov/newsfeed/kmmfeed/FeedViewModel.kt

```diff
@@ -20,8 +20,34 @@ class FeedViewModel(
     dispatcher = dispatcher,
     initialState = State()
 ) {
+    private val searchQueryFlow = MutableStateFlow("")
+
     init {
         loadNextPage()
+        observeSearchQuery()
+    }
+
+    /**
+     * Observer function placed on the search text being updated from the UI.
+     *
+     * The MutableStateFlow, representing the sequence of keystrokes being emitted
+     * by the user, is debounced every 500ms so that downstream services (API,
+     * repository...) are not hit an unnecessary number of times and only when relevant
+     * results can be returned
+     */
+    private fun observeSearchQuery() {
+        launchUiCoroutine {
+            searchQueryFlow
+                .debounce(500)
```

Comment:
this could be in a constant, but is usually outside of the class (looks strange?) - https://stackoverflow.com/questions/44038721/constants-in-kotlin-whats-a-recommended-way-to-create-them

Reply:
Yep we can put this in a private const val at the end of the file 👍

Reply:
done